src/Chemistry.hpp

```diff
@@ -88,15 +93,12 @@ template <typename problem_t> void computeChemistry(amrex::MultiFab &mf, const R
 			// do it in .cpp so that it is not built at compile time for all tests
 			// which would otherwise slow down compilation due to the large RHS file
 			chemburner(chemstate, dt);
+			burnsteps(i, j, k) = chemstate.success;
 
 			if (std::isnan(chemstate.xn[0]) || std::isnan(chemstate.rho)) {
 				amrex::Abort("Burner returned NAN");
 			}
```

Comment:
Don't think this loop is needed? But I must have encountered a case in the past where the burn seemed to be successful but the mass fraction or density returned were NAN, which is why I check for NAN here. But again, this won't work on GPUs all the time. Not sure if 1.) this is still needed and 2.) if yes, how to fix it other than defining new MFs in the way I do to store `chemstate.success`.

Reply:
You could instead have a MultiFab that stores whether it's successful or not (either 0 or 1 or `true` or `false`). Then you can set `false` for any cell that returns NAN.